enhancements/cluster-logging/loki-tokenized-auth-enablement.md

```diff
@@ -0,0 +1,343 @@
+---
+title: loki-tokenized-auth-enablement
+authors:
+  - "@periklis"
+reviewers:
+  - "@cahartma"
+  - "@xperimental"
+  - "@JoaoBraveCoding"
+  - "@btaani"
+approvers:
+  - "@jcantrill"
+  - "@alanconway"
+api-approvers:
+  - "@xperimental"
+creation-date: 2023-10-27
+last-updated: 2023-10-27
+tracking-link:
+  - https://issues.redhat.com/browse/OCPSTRAT-6
+  - https://issues.redhat.com/browse/OCPSTRAT-171 (AWS STS)
+  - https://issues.redhat.com/browse/OCPSTRAT-114 (Azure WIF)
+  - https://issues.redhat.com/browse/OCPSTRAT-922 (GCP WIF)
+  - https://issues.redhat.com/browse/LOG-4540 (AWS & Azure)
+  - https://issues.redhat.com/browse/LOG-4754 (GCP)
+see-also:
+  - "/enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md"
+replaces:
+  - []
+superseded-by:
+  - []
+---
+
+# LokiStack Tokenized Authentication on Cloud Providers
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Many Cloud Providers offer services that allow authentication via short-lived tokens assigned to limited set of privileges. Currently OpenShift supports provisioning tokenized authentication on all major Cloud Providers via the Cloud Credential Operator (CCO), i.e. AWS STS (4.14.0), Azure (4.14.z) and GCP (4.15.0).
+
+This enhancement seeks to bring a unified integration to configure LokiStack with CCO's capabilities to use tokenized authentication for accessing the respective Cloud Providers' object storage service. This improves the UX for consumers of the Loki Operator and in turn the Red Hat OpenShift Logging product.
+
+## Motivation
+
+As previously accomplished by other Red Hat and third-party operators (e.g. OADP Operator, External DNS Operator, AWS EFS CSI Driver Operator) the following proposal seeks to use the CCO's CredentialsRequest API in addition to the OpenShift Console OLM capabilities to detect STS-enabled clusters and manage the whole operator lifecycle.
+
+
+### User Stories
+
+* As a LokiStack administrator I want to off-load STS configuration to CCO so that I do not need to manually manage STS credentials configuration on AWS and in turn a custom LokiStack S3 object storage secret.
+* As a LokiStack administrator I want to off-load Azure WIF configuration to CCO so that I do not need to manually manage WIF credentials configuration on Azure and in turn a custom LokiStack Azure object storage secret.
+* As a LokiStack administrator I want to off-load GCP WIF configuration to CCO so that I do not need to manually manage WIF credentials configuration on GCP and in turn a custom LokiStack GCS object storage secret.
+
```

Comment:
Are we supporting only user stories that use CCO?

Reply:
> Are we supporting only user stories that use CCO?

This verbiage implies the need to use the CCO utility but, aside from this proposal, it should be possible to use any flow that provides the necessary token.  My understanding is CCO simplifies the setup which results in a secret with the needed information.

Reply:
IMHO we should limit our support vector here because an operator bundle with the `features.operators.openshift.io/token-auth-{aws,azure,gcp}` annotation will only work in companion with the CCO workflow on OpenShift. Although we support upstream a setup w/o CCO, I suggest to support only the CCO workflow for OCP subscriptions.